### DISCUSSION TITLE
v-touch-hold in Typescript

### DISCUSSION THREAD

**Discussion by john-** [quasarframework/quasar#16474] [General - Components / Directives / etc] [answered]
This is probably not a Quasar question but I want to make sure there is nothing Quasar related that I am missing.

I am trying to add the `v-touch-hold` directive to my Quasar application and have been unable to work through compile errors:

![quasar_v-touch-hold_template](https://github.com/quasarframework/quasar/assets/2359125/2c7cddb1-9aae-4333-83b6-02d18cdf8f51)

![quasar_v-touch-hold_script](https://github.com/quasarframework/quasar/assets/2359125/8cff7365-e498-48b4-aad7-c753439f7109)

Adding an `any` with this result makes me think I am missing something basic:

![quasar_v-touch-hold_script_2](https://github.com/quasarframework/quasar/assets/2359125/e7be951e-5175-4924-bcf5-a33912a785ec)

I am new to Typescript so currently unable to determine how to map the vs code hint into a fix.

Eventually I want to pass the `xmit` object to the handler but first things first.

**Comment by john-** [comment]
I got it to work even though it is hacky.  I would rather pass the `xmit` object to the handler vs. have the handler retrieve it from the attribute.  However, I could not figure out how to pass it via the handler.
```
  <q-page>
    <q-list>
      <q-item v-for="xmit in xmitsPlayed" :key="xmit.xmit_key">
        <ScannerTransmision :xmit="xmit" :banks="banks" v-bind:xmit-data="JSON.stringify(xmit)" v-touch-hold.mouse="handleHold"/>
      </q-item>
    </q-list>
  </q-page>

[snip]

const handleHold = (obj: any) => {
  let msg = JSON.parse(obj.evt.target.getAttribute('xmit-data'))
  if (!msg) { return }
  msg.replayed = true
  bus.emit('enqueue', msg)
}
```

**Comment by yusufkandemir** [comment, marked as answer]
```ts
import { TouchHoldValue } from 'quasar';

const handleHold: TouchHoldValue = (details) => {
  // ...
}
```

**Reply by john-** [reply]
That worked...thanks!
